nix/mobile/android/maven-and-npm-deps/maven/fetch-maven-deps.sh

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env bash
+
+set -Eeu
+
+# This script takes care of generating/updating the maven-inputs.txt file.
```

Comment:
I'm not even going to try to review this monster. I assume it works and I don't want to touch it EVER.
If it does start causing trouble, I think a better approach would be to `strace` the download process and instead of hardcoding possible sources of packages just find them in the calls that do the requests(dns resolution in this case I guess).

Reply:
I'm willing to timebox giving `strace` a shot, now that all the other problems are sorted out.